core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -158,76 +287,127 @@ public TermsQueryBuilder disableCoord(boolean disableCoord) {
         return this;
     }
 
+    public boolean disableCoord() {
+        return this.disableCoord;
+    }
+
+    public boolean isTermsLookupQuery() {
+        return this.termsLookup.id != null;
+    }
+
+    public TermsQueryBuilder termsLookup(TermsLookupBuilder termsLookup) {
+        this.termsLookup = termsLookup;
+        return this;
+    }
+
+    public TermsLookupBuilder termsLookup() {
+        return this.termsLookup;
+    }
+
     /**
-     * Sets the filter name for the filter that can be used when searching for matched_filters per hit.
->>>>>>> master
+     * Sets the index name to lookup the terms from.
      */
     public TermsQueryBuilder lookupIndex(String lookupIndex) {
-        this.lookupIndex = lookupIndex;
+        this.termsLookup.index = lookupIndex;
```

Comment:
I first did this but really found it pretty ugly. I can revert if you'd like ...